biz.aQute.repository/test/aQute/bnd/repository/osgi/OSGiRepositoryTest.java

```diff
@@ -47,66 +49,80 @@ protected void setUp() throws Exception {
 
 	public void testSimple() throws Exception {
 		try (OSGiRepository r = new OSGiRepository();) {
-			Map<String, String> map = new HashMap<>();
-			map.put("locations", fnx.getBaseURI("/repo/minir5.xml")
-				.toString());
-			map.put("cache", cache.getPath());
-			map.put("max.stale", "10000");
-			r.setProperties(map);
-			Processor p = new Processor();
-			HttpClient httpClient = new HttpClient();
-			httpClient.setCache(cache);
-			httpClient.setRegistry(p);
-			p.addBasicPlugin(httpClient);
-			p.setBase(ws);
-			p.addBasicPlugin(Workspace.createStandaloneWorkspace(p, ws.toURI()));
-			r.setRegistry(p);
-
-			final AtomicInteger tasks = new AtomicInteger();
-
-			p.addBasicPlugin(new ProgressPlugin() {
-
-				@Override
-				public Task startTask(final String name, int size) {
-					System.out.println("Starting " + name);
-					tasks.incrementAndGet();
-					return new Task() {
-
-						@Override
-						public void worked(int units) {
-							System.out.println("Worked " + name + " " + units);
-						}
-
-						@Override
-						public void done(String message, Throwable e) {
-							System.out.println("Done " + name + " " + message);
-						}
-
-						@Override
-						public boolean isCanceled() {
-							return false;
-						}
-					};
-				}
-			});
+			assertTrue(testRepo(r).check());
+		}
+	}
 
-			assertEquals(0, tasks.get());
-			File file = r.get("dummybundle", new Version("0"), null);
-			assertNotNull(file);
-			assertEquals(2, tasks.get()); // 2 = index + file
-			File file2 = r.get("dummybundle", new Version("0"), null);
-			assertNotNull(file2);
-			// second one should not have downloaded
-			assertEquals(2, tasks.get());
-
-			r.refresh();
-			File file3 = r.get("dummybundle", new Version("0"), null);
-			assertNotNull(file3);
-			// second one should not have downloaded
-			assertEquals(2, tasks.get());
+	@SuppressWarnings("deprecation")
+	public void testCompatibilityWithFixedIndexedRepo() throws Exception {
+		try (FixedIndexedRepo r = new FixedIndexedRepo();) {
+			assertTrue(testRepo(r)
```

Comment:
Also need to close Processor returned by testRepo.

Reply:
as before